### PR TITLE
Remove compiled protocols on mix clean. Fixes #3541

### LINKED ISSUE #3541: `mix clean` does not clean compiled protocols
I ran into an issue where my releases contained unexpected protocol files. This is because the `mix clean` task does not clean out the consolidated protocols directory. Should it? Should there instead be a task for `mix clean.protocols` much like there is for `mix deps.clean`?

### PR DESCRIPTION
**Pull request by c-rack**

### CODE REVIEW

File: lib/mix/lib/mix/tasks/compile.protocols.ex [hunk 1]
```diff
@@ -78,4 +78,9 @@ defmodule Mix.Tasks.Compile.Protocols do
         :ok
     end
   end
+
+  def clean do
+    File.rm_rf(default_path)
+  end
+
```
Comment (lexmag) [inline lib/mix/lib/mix/tasks/compile.protocols.ex]: Sorry, nitpicking: extra line.

File: lib/mix/lib/mix/tasks/clean.ex [hunk 2]
```diff
@@ -23,7 +23,7 @@ defmodule Mix.Tasks.Clean do
 
     {opts, _, _} = OptionParser.parse(args, switches: @switches)
 
-    _ = for compiler <- Mix.Tasks.Compile.compilers(),
+    _ = for compiler <- Mix.Tasks.Compile.compilers() ++ [:protocols],
```
Comment (lexmag) [inline lib/mix/lib/mix/tasks/clean.ex]: Would be better if we use `[:protocols] ++`.

**Comment by josevalim** [comment]
We also need a test. :)

**Comment by josevalim** [comment]
We just need to create the directory here and be sure it is removed:

https://github.com/elixir-lang/elixir/blob/master/lib/mix/test/mix/tasks/clean_test.exs#L26

**Comment by c-rack** [comment]
OK, thanks for the feedback. I will update the PR tomorrow. :)

**Comment by c-rack** [comment]
I have added the test and implemented @lexmag's suggestions.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**